src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDataverseCommand.java

```diff
@@ -150,30 +160,38 @@ public Dataverse execute(CommandContext ctxt) throws CommandException {
         }
 
         managedDv.setPermissionModificationTime(new Timestamp(new Date().getTime()));
-        // TODO: save is called here and above; we likely don't need both
-        managedDv = ctxt.dataverses().save(managedDv);
 
-  //      ctxt.index().indexDataverse(managedDv);
         if (facetList != null) {
             ctxt.facets().deleteFacetsFor(managedDv);
+
+            if (!facetList.isEmpty()) {
+                managedDv.setFacetRoot(true);
+            }
+
             int i = 0;
             for (DatasetFieldType df : facetList) {
                 ctxt.facets().create(i++, df, managedDv);
             }
         }
 
         if (inputLevelList != null) {
+            if (!inputLevelList.isEmpty()) {
+                managedDv.addInputLevelsMetadataBlocksIfNotPresent(inputLevelList);
+            }
             ctxt.fieldTypeInputLevels().deleteFacetsFor(managedDv);
-            for (DataverseFieldTypeInputLevel obj : inputLevelList) {
-                obj.setDataverse(managedDv);
-                ctxt.fieldTypeInputLevels().create(obj);
+            for (DataverseFieldTypeInputLevel inputLevel : inputLevelList) {
+                inputLevel.setDataverse(managedDv);
+                ctxt.fieldTypeInputLevels().create(inputLevel);
```

Comment:
We need to validate the input level objects - in the json you can say         "include": false,
        "required": true and the create Dataverse command will add it to the input level table, but you won't be able to create a dataset because the field won't be in the Create Dataset page, but the "required" will be enforced when you try to save it

Reply:
Good point. Added. Thanks.